frontend/components/locale/Selector.vue

```diff
@@ -0,0 +1,106 @@
+<script setup lang="ts">
+import type OverlayPanel from 'primevue/overlaypanel'
+import { useToast } from 'primevue/usetoast'
+
+const toast = useToast()
+const switchLocalePath = useSwitchLocalePath()
+const { t, locale: baseLocale, locales: baseLocales } = useI18n()
+const { languageWasSelectedOrDismissed } = useLocalStorage()
+
+// We go through this rigamarole because the additional information passed into
+// the i18n configuration is present here (namely, flag + name), but it doesn't
+// appear in the typescript configuration.
+interface Option {
+  code: string
+  flag: string
+  name: string
+}
```

Comment:
I might be misunderstanding, but I think you can drop the `computed`s and do:

```ts
const i18n = useI18n();
const locales = i18n.locales as Option[];
const locale = i18n.locale as Option;
const { t } = i18n; 

if (!locales.value.includes(locale.value)) {
  throw createErrorWithRemediation('locale not found', Remediation.FileBug)
}

// Can just use `locale` from here.
```

Reply:
Good question - the `locale` from `i18n` is JUST the `code` - it's a string.